doc/specifications/proposed/content_download/content_download.md

```diff
@@ -0,0 +1,60 @@
+# Content download
+
+> Author: bd@ez.no
+
+## Summary
+This feature covers downloading of Content binary files over HTTP.
+
+It covers:
+- generation of proper links from Twig as well as PHP,
+- permissions checking (the user must be allowed to read the Content)
+- streaming of the file over HTTP
+
+## Components
+
+### Twig helper
+
+> Name: ez_download_path
+> Usage: `ez_download_path( content, 'file' )`
+
+The helper will be used to generate a download link for a given Field/Content.
+
+### Controller action + route
+
+#### Route
+
+Path: `/content/download/{contentId}/{fieldIdentifier}/{fileName}`
+
+Example: ``/content/download/68/file/My file.pdf
+
+Query parameters
+
+- version (optional)
+
+  The version number the file must be downloaded for. Requires the versionview permission.
+  If not specified, the published version is used.
```

Comment:
Maybe the easiest would be to add a dedicated path with explicit version, and maybe even language, that would only be used when explicitly required, and would always require versionView ?

About client side caching, we've concluded iirc that it doesn't apply that much to binary files (it does to images). But the files should be cachable by a reverse proxy at the very least. With Varnish, we should be able to leverage user-hash as well as location id based tagging so that the file can be expired when a new version of the content is published.

What do you think ?